python-package/lightgbm/basic.py

```diff
@@ -294,6 +294,51 @@ def to_string(x):
     return ' '.join(pairs)
 
 
+def _separate_metrics_list(metrics_list):
+    """Separate built-in from callable evaluation metrics."""
+    metrics_callable = [f for f in metrics_list if callable(f)]
+    metrics_builtin = [m for m in metrics_list if isinstance(m, str)]
+    return metrics_callable, metrics_builtin
+
+
+def _concat_params_metrics(params, metrics_builtin):
+    """Concatenate metric from params (or default if not provided in params) and eval_metric."""
+    params = deepcopy(params)
+    params_metric = deepcopy(params.get('metric'))
+    params_metric = [params_metric] if isinstance(params_metric, (str, type(None))) else params_metric
+    params_metric = [e for e in metrics_builtin if e not in params_metric] + params_metric
+    params_metric = [metric for metric in params_metric if metric is not None]
+    params['metric'] = params_metric
+    return params
+
+
+def _concat_metric_feval_callables(metrics_callable, feval_callable):
```

Comment:
Based on https://github.com/microsoft/LightGBM/pull/5052#issuecomment-1065949317, this PR should only contain changes related to callable objective functions, not anything related to `metric`.

Can you please remove this and other changes that are specific to `metric` from this PR?

As you've said in https://github.com/microsoft/LightGBM/pull/5052#issuecomment-1067492446, a separate PR will deal with changes related to `metric`.

Reply:
copy that